### PR TITLE
grype-v0.65.2-trivy-0.44.1-syft-v0.87.1-upgrade

### PR DESCRIPTION
**Pull request by rmedvedo**
## Description

Grype version upgraded. Trivy and Syft upgraded because of compatibility with Grype's new version.

## Type of Change

[ ] Bug Fix
[ ] New Feature
[ ] Breaking Change
[*] Refactor
[ ] Documentation
[ ] Other (please describe)

## Checklist

- [*] I have read the [contributing guidelines](/CONTRIBUTING.md)
- [*] Existing issues have been referenced (where applicable)
- [*] I have verified this change is not present in other open pull requests
- [*] Functionality is documented
- [*] All code style checks pass
- [*] New code contribution is covered by automated tests
- [*] All new and existing tests pass

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -19,8 +19,7 @@ linters-settings:
     replace-allow-list:
       # replace containerd GHSA-5ffw-gxpp-mxpf
       - github.com/containerd/containerd
-      # trivy requires these replacements
-      - oras.land/oras-go
+      - github.com/docker/docker
```
Comment (FrimIdan) [inline .golangci.yml:22]: please add a comment for the reason of the replace

**Review by FrimIdan — CHANGES_REQUESTED** [review]

File: cli/go.mod [hunk 2]
```diff
@@ -422,8 +434,7 @@ replace (
 // Replace containerd due to GHSA-5ffw-gxpp-mxpf
 replace github.com/containerd/containerd => github.com/containerd/containerd v1.6.6
 
-// Replace these for trivy
-replace oras.land/oras-go => oras.land/oras-go v1.1.1
```
[the remark this answers is not in the document]
Comment (rmedvedo) [reply cli/go.mod:426]: If it not removed and oras not upgraded there will be issues with mismatching versions of other dependencies.

File: shared/pkg/analyzer/syft/syft.go [hunk 3]
```diff
@@ -118,6 +120,12 @@ func (a *Analyzer) setError(res *analyzer.Results, err error) {
 	a.resultChan <- res
 }
 
-func getImageHash(sbom syft_sbom.SBOM, src string) string {
-	return image_helper.GetHashFromRepoDigest(sbom.Source.ImageMetadata.RepoDigests, src)
+func getImageHash(sbom syft_sbom.SBOM, src string) (string, error) {
+	if metadata, ok := sbom.Source.Metadata.(source.StereoscopeImageSourceMetadata); ok {
```
[the remark this answers is not in the document]
Comment (rmedvedo) [reply shared/pkg/analyzer/syft/syft.go:124]: This is the only type with repoDigests.

File: .golangci.yml (hunk 1 above)
Comment (rmedvedo) [reply .golangci.yml:23]: Added

File: shared/pkg/analyzer/syft/syft.go [hunk 4]
```diff
@@ -17,7 +17,6 @@ package syft
 
 import (
 	"fmt"
-
```
[the remark this answers is not in the document]
Comment (rmedvedo) [reply shared/pkg/analyzer/syft/syft.go]: undone

File: cli/go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (rmedvedo) [reply cli/go.mod:426]: The replace changed according to Trivy.

File: shared/pkg/analyzer/syft/syft.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (rmedvedo) [reply shared/pkg/analyzer/syft/syft.go:125]: Changed to switch/case as requested.

File: shared/pkg/analyzer/syft/syft.go [hunk 5]
```diff
@@ -61,11 +61,13 @@ func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
 	a.logger.Infof("Called %s analyzer on source %s", a.name, src)
 	// TODO platform can be defined
 	// https://github.com/anchore/syft/blob/b20310eaf847c259beb4fe5128c842bd8aa4d4fc/cmd/syft/cli/options/packages.go#L48
-	input, err := source.ParseInput(src, "")
+	detection, err := source.Detect(userInput, source.DefaultDetectConfig())
```
Comment (FrimIdan) [inline shared/pkg/analyzer/syft/syft.go:64]: I think that you should use `src` and not `userInput`

File: shared/pkg/analyzer/trivy/trivy.go [hunk 6]
```diff
@@ -92,10 +92,10 @@ func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
 				Scanners: []trivyTypes.Scanner{}, // Disable all security checks for SBOM only scan
 			},
 			ReportOptions: trivyFlag.ReportOptions{
-				Format:       "cyclonedx", // Cyconedx format for SBOM so that we don't need to convert
-				ReportFormat: "all",       // Full report not just summary
-				Output:       &output,     // Save the output to our local buffer instead of Stdout
-				ListAllPkgs:  true,        // By default Trivy only includes packages with vulnerabilities, for full SBOM set true.
+				Format:       "cyclonedx",     // Cyconedx format for SBOM so that we don't need to convert
+				ReportFormat: "all",           // Full report not just summary
+				Output:       output.String(), // Save the output to our local buffer instead of Stdout
```
Comment (FrimIdan) [inline shared/pkg/analyzer/trivy/trivy.go:97]: This not gonna work, it looks like trivy changed it from a report into a buffer to something else. Need to understand how it possible to get the output into a buffer again (or find an alternative, I don't like to create a temp file and write it there but maybe we will not have any other option)

File: shared/pkg/analyzer/trivy/trivy.go [hunk 7]
```diff
@@ -92,10 +92,10 @@ func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
...
+				ListAllPkgs:  true,            // By default Trivy only includes packages with vulnerabilities, for full SBOM set true.
 			},
 			VulnerabilityOptions: trivyFlag.VulnerabilityOptions{
 				VulnType: trivyTypes.VulnTypes, // Trivy disables analyzers for language packages if VulnTypeLibrary not in VulnType list
```
Comment (FrimIdan) [inline shared/pkg/analyzer/trivy/trivy.go:101]: Please add the following in the trivy options
```
			ImageOptions: trivyFlag.ImageOptions{
				ImageSources: types.AllImageSources,
			},
```

File: shared/pkg/scanner/trivy/scanner.go [hunk 8]
```diff
@@ -126,11 +126,11 @@ func (a *Scanner) createTrivyOptions(output *bytes.Buffer, userInput string) (tr
...
-			ReportFormat: "all",      // Full report not just summary
-			Output:       output,     // Save the output to our local buffer instead of Stdout
-			ListAllPkgs:  false,      // Only include packages with vulnerabilities
-			Severities:   severities, // All the severities from the above
+			Format:       "json",          // Trivy's own json format is the most complete for vuls
+			ReportFormat: "all",           // Full report not just summary
+			Output:       output.String(), // Save the output to our local buffer instead of Stdout
+			ListAllPkgs:  false,           // Only include packages with vulnerabilities
+			Severities:   severities,      // All the severities from the above
 		},
```
Comment (FrimIdan) [inline shared/pkg/scanner/trivy/scanner.go:134]: Please add the following in the trivy options
```
			ImageOptions: trivyFlag.ImageOptions{
				ImageSources: types.AllImageSources,
			},
```

File: shared/pkg/scanner/trivy/scanner.go [hunk 9]
```diff
@@ -126,11 +126,11 @@ func (a *Scanner) createTrivyOptions(output *bytes.Buffer, userInput string) (tr
 			},
 		},
 		ReportOptions: trivyFlag.ReportOptions{
-			Format:       "json",     // Trivy's own json format is the most complete for vuls
-			ReportFormat: "all",      // Full report not just summary
-			Output:       output,     // Save the output to our local buffer instead of Stdout
-			ListAllPkgs:  false,      // Only include packages with vulnerabilities
-			Severities:   severities, // All the severities from the above
+			Format:       "json",          // Trivy's own json format is the most complete for vuls
```
Comment (FrimIdan) [inline shared/pkg/scanner/trivy/scanner.go:129]: please use `trivyTypes.FormatJSON`

File: shared/pkg/analyzer/trivy/trivy.go [hunk 10]
```diff
@@ -92,10 +92,10 @@ func (a *Analyzer) Run(sourceType utils.SourceType, userInput string) error {
 				Scanners: []trivyTypes.Scanner{}, // Disable all security checks for SBOM only scan
 			},
 			ReportOptions: trivyFlag.ReportOptions{
-				Format:       "cyclonedx", // Cyconedx format for SBOM so that we don't need to convert
-				ReportFormat: "all",       // Full report not just summary
-				Output:       &output,     // Save the output to our local buffer instead of Stdout
-				ListAllPkgs:  true,        // By default Trivy only includes packages with vulnerabilities, for full SBOM set true.
+				Format:       "cyclonedx",     // Cyconedx format for SBOM so that we don't need to convert
```
Comment (FrimIdan) [inline shared/pkg/analyzer/trivy/trivy.go:95]: please use `trivyTypes.FormatCycloneDX`

File: charts/kubeclarity/values.yaml [hunk 11]
```diff
@@ -244,8 +244,8 @@ kubeclarity-grype-server:
 
   ## Docker Image values.
   docker:
-    imageRepo: "ghcr.io/openclarity"
-    imageTag: "v0.2.0"
+    imageRepo: "rmedvedo173"
+    imageTag: "0.0.1"
```
Comment (FrimIdan) [inline charts/kubeclarity/values.yaml:248]: need to revert after testing

**Review by akpsgit — APPROVED** [review]

**Review by akpsgit — APPROVED** [review]

**Review by FrimIdan — APPROVED** [review]

**Pull request merged**